index/scorch/scorch.go

```diff
@@ -117,9 +121,30 @@ func NewScorch(storeName string,
 	return rv, nil
 }
 
+func (s *Scorch) waterMark() uint64 {
+	s.memLock.Lock()
+	w := s.highWaterMark
+	s.memLock.Unlock()
+	return w
+}
+
+func (s *Scorch) incrWaterMark() {
+	s.memLock.Lock()
+	s.highWaterMark++
+	s.memLock.Unlock()
+}
+
+func (s *Scorch) decrWaterMark() {
+	s.memLock.Lock()
+	s.highWaterMark--
+	s.memLock.Unlock()
+}
+
 func (s *Scorch) fireEvent(kind EventKind, dur time.Duration) {
 	if s.onEvent != nil {
+		s.incrWaterMark()
 		s.onEvent(Event{Kind: kind, Scorch: s, Duration: dur})
+		s.decrWaterMark()
```

Comment:
naming wise, this doesn't look like a water mark?

perhaps other thoughts of that this looks like "inMiddleOfEventCallback"?

or, actually, the reason why this is here is the even handler (the app-herder) might end up pausing during the event callback (e.g., it's stepping on the breaks), so maybe the "water mark" counter is more like "pauseCount" or something like that?